### PR TITLE
[Merged by Bors] - chore(ModelTheory/Complexity): move quantifier complexity into a new file

### PR DESCRIPTION
**Pull request by awainverse**
Several important files in the model theory library touch on quantifier complexity of formulas, an idea that is not yet used for anything else in `mathlib`.
This PR moves all quantifier complexity content to a new file, `ModelTheory/Complexity` and improves the documentation of these concepts.
This will hopefully make these basic files more legible and easier to refactor.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/ModelTheory/Complexity.lean [hunk 1]
```diff
@@ -0,0 +1,322 @@
+/-
+Copyright (c) 2021 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Anderson
+-/
+import Mathlib.ModelTheory.Satisfiability
+
+/-!
+# Quantifier Complexity
+This file defines quantifier complexity of first-order formulas, and constructs prenex normal forms.
```
Comment (YaelDillies) [inline Mathlib/ModelTheory/Complexity.lean:10]:
```suggestion

This file defines quantifier complexity of first-order formulas, and constructs prenex normal forms.
```

File: Mathlib/ModelTheory/Complexity.lean [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+## Main Definitions
```
Comment (YaelDillies) [inline Mathlib/ModelTheory/Complexity.lean:12]:
```suggestion
## Main Definitions

```

File: Mathlib/ModelTheory/Complexity.lean [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+* `FirstOrder.Language.BoundedFormula.IsAtomic` defines atomic formulas - those which are
+constructed only from terms and relations.
+* `FirstOrder.Language.BoundedFormula.IsQF` defines quantifier-free formulas - those which are
+constructed only from atomic formulas and boolean operations.
+* `FirstOrder.Language.BoundedFormula.IsPrenex` defines when a formula is in prenex normal form -
+when it consists of a series of quantifiers applied to a quantifier-free formula.
+* `FirstOrder.Language.BoundedFormula.toPrenex` constructs a prenex normal form of a given formula.
+
+
+## Main Results
```
Comment (YaelDillies) [inline Mathlib/ModelTheory/Complexity.lean:22]:
```suggestion
## Main Results

```

File: Mathlib/ModelTheory/Complexity.lean [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+variable {α : Type u'} {β : Type v'} {γ : Type*}
+variable {n l : ℕ} {φ ψ : L.BoundedFormula α l} {θ : L.BoundedFormula α l.succ}
+variable {v : α → M} {xs : Fin l → M}
+
+open FirstOrder Structure Fin
+
+namespace BoundedFormula
+
+/-- An atomic formula is either equality or a relation symbol applied to terms.
+  Note that `⊥` and `⊤` are not considered atomic in this convention. -/
```
Comment (YaelDillies) [inline Mathlib/ModelTheory/Complexity.lean:45]: We are trying not to indent docstrings by default
```suggestion
Note that `⊥` and `⊤` are not considered atomic in this convention. -/
```

**Review by YaelDillies — APPROVED** [review]
Looks very sensible!

**Comment by YaelDillies** [comment]
maintainer merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**